File: backend/app/schemas/stock_picker.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import List


class StockPoolItemBase(BaseModel):
    symbol: str
    name: str | None = None
    score: float | None = None
    reason: str | None = None


class StockPoolItemRead(StockPoolItemBase):
    model_config = ConfigDict(from_attributes=True)

    id: int


class StockPoolBase(BaseModel):
    pool_id: str
    picker_id: str
    name: str
    is_builtin_weekly: bool = False


class StockPoolCreate(StockPoolBase):
    pass


class StockPoolRead(StockPoolBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    generated_at: datetime
    expires_at: datetime | None = None
    items: List[StockPoolItemRead] = []


class PickerRunBase(BaseModel):
    run_id: str
    picker_id: str
    status: str = "pending"
    result_count: int | None = None
    logs: str | None = None


class PickerRunRead(PickerRunBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: datetime
    finished_at: datetime | None = None


class NotificationSettingsRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    weekly_picker_push: bool
    updated_at: datetime


class NotificationSettingsUpdate(BaseModel):
    weekly_picker_push: bool


class RunPickerRequest(BaseModel):
    picker_id: str


class WeeklyPickerSummary(BaseModel):
    has_new_weekly: bool
    pool: StockPoolRead | None = None
    generated_at: datetime | None = None
    item_count: int = 0

File: frontend/src/api/stockPicker.ts
import { delay } from './mock/utils'
import { DEMO_STOCK_POOLS, DEMO_PICKER_RUNS } from './mock/demoData'

export interface StockPoolItem {
  id: number
  symbol: string
  name?: string
  score?: number
  reason?: string
}

export interface StockPool {
  id: number
  pool_id: string
  picker_id: string
  name: string
  is_builtin_weekly: boolean
  generated_at: string
  expires_at?: string
  items: StockPoolItem[]
}

export interface PickerRun {
  id: number
  run_id: string
  picker_id: string
  status: string
  result_count?: number
  logs?: string
  created_at: string
  finished_at?: string
}

export interface WeeklySummary {
  has_new_weekly: boolean
  pool?: StockPool
  generated_at?: string
  item_count: number
}

export interface NotificationSettings {
  id: number
  weekly_picker_push: boolean
  updated_at: string
}

export const stockPickerApi = {
  runPicker(_picker_id: string): Promise<StockPool> {
    return delay(500).then(() => DEMO_STOCK_POOLS[0] as StockPool)
  },
  listPools(_picker_id?: string): Promise<StockPool[]> {
    return delay(300).then(() => DEMO_STOCK_POOLS as StockPool[])
  },
  getPool(_pool_id: string): Promise<StockPool> {
    return delay(300).then(() => DEMO_STOCK_POOLS[0] as StockPool)
  },
  listRuns(_picker_id?: string): Promise<PickerRun[]> {
    return delay(300).then(() => DEMO_PICKER_RUNS as PickerRun[])
  },
  getWeeklySummary(): Promise<WeeklySummary> {
    return delay(300).then(() => ({
      has_new_weekly: true,
      pool: DEMO_STOCK_POOLS[0] as StockPool,
      generated_at: new Date().toISOString(),
      item_count: 3,
    }))
  },
  getNotificationSettings(): Promise<NotificationSettings> {
    return delay(300).then(() => ({
      id: 1,
      weekly_picker_push: true,
      updated_at: new Date().toISOString(),
    }))
  },
  updateNotificationSettings(payload: { weekly_picker_push: boolean }): Promise<NotificationSettings> {
    return delay(400).then(() => ({
      id: 1,
      weekly_picker_push: payload.weekly_picker_push,
      updated_at: new Date().toISOString(),
    }))
  },
}
